pennylane_lightning/core/src/simulators/lightning_kokkos/gates/BasicGateFunctors.hpp

```diff
@@ -600,7 +651,7 @@ template <class PrecisionT, class FuncT> class applyNC2Functor {
     template <class ExecutionSpace>
     applyNC2Functor([[maybe_unused]] ExecutionSpace exec,
                     Kokkos::View<Kokkos::complex<PrecisionT> *> arr_,
-                    std::size_t num_qubits,
+                    const std::size_t num_qubits,
```

Comment:
Is it necessary to set `num_qubits` as `const` for the argument?

Reply:
I will keep consistent with the original file that e.g. `applyNC2Functor` `applyNC1Functor` etc. does not use const, just `std::size_t num_qubits`, but I will keep the `const std::size_t num_qubits` in specific functions `applyPauliY`. This could be revisited in a later PR.